### PR TITLE
B js doc theme

### PR DESCRIPTION
**Pull request by cwattrus**
This is the new Braintree JSDoc template. No longer using Minami theme. 

**Added**
Highlight.js - Code highlighting
B Mono - Braintree font
Algolia - Search
Mixpanel - Analytics

**Removed**
Minami
Prettify

### CODE REVIEW

**Comment by crookedneighbor** [comment]
Q: was anything form minami kept? If not, would it make more sense to create a new repo than to retain the git history of the minami repo?

File: package.json [hunk 1]
```diff
@@ -1,22 +1,24 @@
 {
   "name": "jsdoc-template",
```
Comment (crookedneighbor) [inline package.json:2]: Do we consider this template open source? Should we give it a name and publish it to npm? 

We could also publish it under the `@braintree` namespace.

File: publish.js [hunk 2]
```diff
@@ -298,15 +298,16 @@ function buildMemberNav(items, itemHeading, itemsSeen, linktoFn) {
             var members = find({kind:'member', memberof: item.longname});
 
             if ( !hasOwnProp.call(item, 'longname') ) {
-                itemsNav += '<li>' + linktoFn('', item.name);
+                itemsNav += "<li id='" + item.name.replace('/','_') + "-nav'>" + linktoFn('', item.name);
                 itemsNav += '</li>';
             } else if ( !hasOwnProp.call(itemsSeen, item.longname) ) {
-                itemsNav += '<li>' + linktoFn(item.longname, item.name.replace(/^module:/, ''));
+              // replace '/' in url to match ID in some section
```
Comment (crookedneighbor) [inline publish.js]: Indentation of comment is off here.

File: publish.js [hunk 3]
```diff
@@ -298,15 +298,16 @@ function buildMemberNav(items, itemHeading, itemsSeen, linktoFn) {
             var members = find({kind:'member', memberof: item.longname});
 
             if ( !hasOwnProp.call(item, 'longname') ) {
-                itemsNav += '<li>' + linktoFn('', item.name);
+                itemsNav += "<li id='" + item.name.replace('/','_') + "-nav'>" + linktoFn('', item.name);
```
Comment (crookedneighbor) [inline publish.js]: Generally we use `'` for JS and `"` for html props.

**Review by crookedneighbor — COMMENTED** [review]
Since this is our code now (instead of slightly modified minami), should we be linting with eslint and our config?

**Comment by crookedneighbor** [comment]
RE: Since this is our code now (instead of slightly modified minami), should we be linting with eslint and our config?

Let's merge this in as is and we can add an issue to use eslint and our eslint-config.

**Comment by crookedneighbor** [comment]
Talked separately with @cwattrus about this. We're going to create cards for my comments, since they don't need to block this from being merged.

File: static/scripts/pagelocation.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
+$( document ).ready(function() {
+  // If an anchor hash is in the URL highlight the menu item
+  highlightActiveHash();
+  // If a specific page section is in the URL highlight the menu item
+  highlightActiveSection();
+
+  // If a specific page section is in the URL scroll that section up to the top
+  $("nav").scrollTop($("#" + getCurrentSectionName() + "-nav").position().top);
+
+  // function to scroll to anchor when clicking an anchor linl
```
Comment (lilaconlee) [inline static/scripts/pagelocation.js:14]: linl => link 
hugely unimportant

File: static/scripts/pagelocation.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  $('a[href*="#"]:not([href="#"])').click(function() {
+    if (location.pathname.replace(/^\//,'') == this.pathname.replace(/^\//,'') && location.hostname == this.hostname) {
```
Comment (lilaconlee) [inline static/scripts/pagelocation.js]: I would use `===` instead of `==` here

File: static/scripts/pagelocation.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  $("#" + pageId + "-nav").addClass("active");
+}
+
+function getCurrentSectionName() {
+  var path = window.location.pathname;
+  var pageUrl = path.split("/").pop();
+  
+  var sectionName = pageUrl.substring(0, pageUrl.indexOf("."));
+
+  // remove the wodr module- if its in the url
```
Comment (lilaconlee) [inline static/scripts/pagelocation.js:68]: wodr => word

File: static/styles/jsdoc-default.css [hunk 7]
```diff
@@ -411,77 +558,87 @@ h4.name {
 }
 
 .prettyprint.linenums ol {
-   padding-left: 0
+  padding-left: 0
```
Comment (lilaconlee) [inline static/styles/jsdoc-default.css]: 561, 565, & 570 could use some semicolons

File: static/styles/jsdoc-default.css [hunk 8]
```diff
@@ -411,77 +558,87 @@ h4.name {
...
+  padding-bottom: 0;
 }
 
 dl.param-type {
   margin-top: 5px;
 }
 
-.param-type dt, .param-type dd {
+.param-type dt,
+.param-type dd {
```
Comment (lilaconlee) [inline static/styles/jsdoc-default.css]: 642, 646, & 650 could use semicolons as well

File: static/styles/jsdoc-default.css [hunk 9]
```diff
@@ -493,103 +650,146 @@ dl.param-type {
...
+.tag-source:first-child {
+  border-bottom: 1px solid rgba(28, 160, 224, 1);
+}
+
+
 /* navicon button */
+
 .navicon-button {
-  display: none;
+  /*display: none;*/
```
Comment (lilaconlee) [inline static/styles/jsdoc-default.css]: I'd prefer we just remove instead of commenting out, but I'm not passionate about it

**Comment by lilaconlee** [comment]
So awesome! I'm 🐈  once we get some linting.

**Comment by crookedneighbor** [comment]
Are tables with 3 columns (instead of the normal 4) styled correctly?

screen shot 2016-10-31 at 2 30 43 pm

**Comment by cwattrus** [comment]
@lilaconlee @crookedneighbor so have made some changes + Blade's inclusion of the linted things. Is there anything else required before we merge this in?

File: package.json (hunk 1 above)
Comment (crookedneighbor) [reply package.json:2]: I think we should give this a proper name, but it can wait for after this PR.

File: package.json [hunk 10]
```diff
@@ -1,22 +1,28 @@
...
-  "version": "1.2.6",
-  "description": "Clean and minimal JSDoc 3 Template / Theme, forked from minami",
+  "version": "2.0.0",
```
Comment (crookedneighbor) [inline package.json:3]: :+1: on bumping the version to 2.0.0 I'll tag it once it is merged in

**Comment by crookedneighbor** [comment]
If all of @lilaconlee's comments have been addressed, :+1:

**Comment by lilaconlee** [comment]
🐈  looks good to go!!

**Pull request merged**